test/extended/cli/admin.go

```diff
@@ -0,0 +1,517 @@
+		// o.Expect(err).NotTo(o.HaveOccurred())
+		// out, err = ocns.WithToken(token).Run("get").Args("user/~").Output()
+		// o.Expect(err).NotTo(o.HaveOccurred())
+		// o.Expect(out).To(o.ContainSubstring(fmt.Sprintf("system:serviceaccount:%s:my-sa-name", ocns.Namespace())))
+
+		// add a new labeled token and ensure the label stuck
+		o.Expect(ocns.Run("sa", "new-token").Args("my-sa-name", "--labels=mykey=myvalue,myotherkey=myothervalue").Execute()).To(o.Succeed())
+		out, err = ocns.Run("get").Args("secrets", "--selector=mykey=myvalue").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("my-sa-name"))
+		out, err = ocns.Run("get").Args("secrets", "--selector=myotherkey=myothervalue").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("my-sa-name"))
+		out, err = ocns.Run("get").Args("secrets", "--selector=mykey=myvalue,myotherkey=myothervalue").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("my-sa-name"))
+
+		ocns.Run("delete").Args("sa/my-sa-name").Execute()
+	})
+
+	g.It("user-creation", func() {
+		o.Expect(oc.Run("create", "user").Args("test-cmd-user").Execute()).To(o.Succeed())
+		o.Expect(oc.Run("create", "identity").Args("test-idp:test-uid").Execute()).To(o.Succeed())
+		o.Expect(oc.Run("create", "useridentitymapping").Args("test-idp:test-uid", "test-cmd-user").Execute()).To(o.Succeed())
+
+		out, err := oc.Run("describe").Args("identity", "test-idp:test-uid").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("test-cmd-user"))
+		out, err = oc.Run("describe").Args("user", "test-cmd-user").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("test-idp:test-uid"))
+
+		oc.Run("delete").Args("user/test-cmd-user").Execute()
+		oc.Run("delete").Args("identity/test-idp:test-uid").Execute()
+		oc.Run("delete").Args("useridentitymapping/test-idp:test-uid").Execute()
+	})
+
+	g.It("images", func() {
+		stableBusyboxPath := exutil.FixturePath("testdata", "stable-busybox.yaml")
+		o.Expect(oc.Run("create").Args("-f", stableBusyboxPath).Execute()).To(o.Succeed())
+
+		out, err := oc.Run("adm", "top", "images").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.MatchRegexp(`sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6\W+default/busybox \(latest\)\W+<none>\W+<none>\W+yes\W+653\.4KiB`))
+		out, err = oc.Run("adm", "top", "imagestreams").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.MatchRegexp(`default/busybox\W+653\.4KiB\W+1\W+1`))
+
+		oc.Run("delete").Args("-f", stableBusyboxPath).Execute()
+	})
+
+	// TODO (soltysh): sync with Standa and figure out if we can get these
+	// enabled back, they were all disabled in admin.sh:
+
+	// 	os::test::junit::declare_suite_start "cmd/admin/rolebinding-allowed"
+	// # Admin can bind local roles without cluster-admin permissions
```

Comment:
You could create a synthetic user by creating a `User` and `OAuthAccessToken` objects, in which case you would use `oc login -t <token>` to log in as that user.

Reply:
I think the non-existence of the `local-admin` user is probably the only reason these tests were disable, right?

Reply:
Unfortunately, the token usage is not supported, yet. See my other comment earlier in the file. I will work on getting it work, though. But it might be a bit later.